### ISSUE TITLE
[web:a11y] Link widget breaks when ensureSemantics is enabled

### ISSUE THREAD

**Issue by johnpryan** [flutter/flutter#143164]
Enabling semantics using `SemanticsBinding.instance.ensureSemantics()` causes the Link widget to break.

Steps to reproduce
1. Create a new Flutter project and copy the contents below
2. Run the app in Chrome and observe that the link doesn't work
3. Comment out `SemanticsBinding.instance.ensureSemantics()`, re-run and observe that the link now works.

<details>
<summary>Code snippet</summary>

main.dart:
```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';
import 'package:go_router/go_router.dart';
import 'package:url_launcher/link.dart';
import 'package:flutter_web_plugins/url_strategy.dart';


void main() {
  usePathUrlStrategy();
  runApp(const MyApp());
  SemanticsBinding.instance.ensureSemantics();
}

final GoRouter _router = GoRouter(
  routes: <RouteBase>[
    GoRoute(
      path: '/',
      builder: (BuildContext context, GoRouterState state) {
        return const HomeScreen();
      },
      routes: <RouteBase>[
        GoRoute(
          path: 'details-screen',
          builder: (BuildContext context, GoRouterState state) {
            return const DetailsScreen();
          },
        ),
      ],
    ),
  ],
);

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp.router(
      routerConfig: _router,
    );
  }
}

class HomeScreen extends StatelessWidget {
  const HomeScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Home Screen')),
      body: Center(
        child: Link(
          builder: (context, followLink) {
            return ElevatedButton(
              onPressed: followLink,
              child: const Text('Go to the Details screen'),
            );
          },
          uri: Uri.parse('/details-screen'),
        ),
      ),
    );
  }
}

class DetailsScreen extends StatelessWidget {
  const DetailsScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Details Screen')),
      body: Center(
        child: ElevatedButton(
          onPressed: () => context.go('/'),
          child: const Text('Go back to the Home screen'),
        ),
      ),
    );
  }
}
```

pubspec.yaml
```yaml
name: link_widget_bug
description: "A new Flutter project."
publish_to: 'none'
version: 0.1.0

environment:
  sdk: '>=3.4.0-82.0.dev <4.0.0'

dependencies:
  flutter:
    sdk: flutter
  flutter_web_plugins:
    sdk: flutter
  go_router: any
  url_launcher: any

dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_lints: ^3.0.0

flutter:
  uses-material-design: true

```
</details>

**Comment by johnpryan** [comment]
cc: @mdebbar

**Comment by emma-roudabush-disney** [comment]
I am also seeing this issue--we lose all link preview and context menu options

**Comment by ss1565** [comment]
I am also seeing this issue.

**Comment by pedromassango** [comment]
This is still happing
